lib/unicode.ml

```diff
@@ -121,30 +121,44 @@ let classify =
 
 exception End_of_input
 
+(* Guillaume Melquiond proposes this alternative code:
```

Comment:
After a careful review, this code looks correct indeed. @ejgallego, can you update this PR with Guillaume's code for testing? Thanks!